### ISSUE TITLE
Why icon Button is not more clickable after closing DialogFramework ? 

### ISSUE THREAD

**Issue by schlebe** [primefaces/primefaces#3373]
I have posted this question on forum but even kukeltje (the more activ PrimeFaces expert) doesn't have answered.

I develop now using Primefaces 6.1.14.

I think that the best thing is to follow my post on forum in this address [https://forum.primefaces.org/viewtopic.php?f=3&t=54492](https://forum.primefaces.org/viewtopic.php?f=3&t=54492)

In short, the problem is that I have developped a solution (workaround) to intercept DialogFramework closing. When user click on Close icon, the process intercept "hide" function display a Confirmation dialog in which user is invited to continue or abort closing of dialog.

In fact the Framework dialog is a dialog to input some values.
When closing dialog, the program detect is user has made some changes or not.
If user has made no change, the dialog framework is closed (or hidden).
If user has made some changes, custom Javascript display a Confirmation dialog (<p:dialog>) in which user can answer YES to confirm closing of dialog without saving or NO to cancel closing.

The problem is than when I click to YES to close Confirmation dialog and DialogFrame work to display again base Form the icon on which I have clicked to display the Dialog framework seems to be locked.

All the code is given in forum post.

The issue is that icon-button is locked !

**Comment by cnsgithub** [comment]
Why so complicated. I would avoid such complex JS workarounds if possible. 

Example how you could do it, if i understood your use case correctly. You just need to add a function to check if the values have been changed.

```xml

<p:commandButton onclick="PF('dlg11').show();" value="Open"/>
	<p:dialog widgetVar="dlg11" onHide="PF('dlg11confirmation').show();">
		<p:inputText value="333"/>
	</p:dialog>
	<p:dialog widgetVar="dlg11confirmation">
		Are you sure?
		<p:commandButton type="button" onclick="PF('dlg11confirmation').hide();" value="Yes"/>
		<p:commandButton type="button" onclick="PF('dlg11confirmation').hide();PF('dlg11').show();" value="No"/>
	</p:dialog>
```

**Comment by schlebe** [comment]
I have read your explanation that is a tricky solution that use that fact that dialog are not really closed but only hidden.

Your point of view is interesting; but one functionnality is missing.

The DialogFramework is hidden before the Confirmation dialog is shown and that is not acceptable for me. If the process ask if user will close the DialogFramework or not, the DialogFramework must be visible.

And that is not all. If I accept that DialogFramework is hidden when Confirmation dialog is displayed, the implementation when dlg1 is a DialogFramework is more complex.

First of all, we have chance, "onHide" event has been implemented for DialogFramework.  So, no problem to use "onHide" event.

But PF('dialogFramework') is not so simple that what you write for <p:dialog> because it is not possible to specifiy a widgetvar name to a dialogFramework (Primefaces 6.1.14).

I have posted a question on StackOverflow that expose this problem 
[https://stackoverflow.com/questions/48907227/on-primefaces-in-a-dialogframework-how-to-intercept-pdialog-close-event-and](https://stackoverflow.com/questions/48907227/on-primefaces-in-a-dialogframework-how-to-intercept-pdialog-close-event-and) You can vote UP :-)

Another question is when I have found the DialogFramework widgetvar, how can I make it modal again ?

**Comment by schlebe** [comment]
@cnsgithub: I have used your technics, and I have replaced

`wgDialog.hide = function() { displayConfirmDialog();
`

where wgDialog is the DialogFramework widged variable.

by ...

```
wgDialog.closeIcon.off('click'); 
wgDialog.closeIcon.click(function() { displayConfirmDialog(); });
```
this work well, 

But I have the same problem that I have with icon/button that keep blocked and unclickable !

Sorry, I have forgotten to change the method call to close the DialogFramework.

Before, I called

```
wgDialog.jq.hide();
wgDialog.disableModality();
```

and now I can call directly

`wgDialog.hide();`

When I do that, DialogFramework is correctly closed ... and button in original Form call be clicked a new time.

The trick with your proposal is that Dialog.hide() function is not modified and that I can use it normally when I will hide Dialog.

Thanks for your help.

Now I have a complete solution to intercept DialogFramework before the dialog is closed to display a Confirmation <p:dialog> to advise user that some changes has not been saved.
